### PR TITLE
Update mojo-parent to 76, Maven 3.6.3, JDK 8 and GitHub Actions

### PR DESCRIPTION
**Pull request by EagleErwin**
This Pull Request updates the mojo-parent version to 76. Together with this, the Maven API is changed from 2.2.1 to 3.6.3 and the minimum Java version is changed from 6 to 8.

It also migrates from Travis CI to GitHub Actions CI.

PR #119 broke integration test rpm-dirmode. For now, I disabled this test. It should be enabled again when issues #118 and #125 are solved.

### CODE REVIEW

File: src/it/rpm-dirmode/verify.groovy [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// Disable this test for now
+System.exit(0)
```
Comment (slawekjaranowski) [inline src/it/rpm-dirmode/verify.groovy:2]: Don't do it in this way.

You can add `pomExcludes` to m-invoker-p or create a file: `selector.groovy` with body 
```
return false
```

https://maven.apache.org/plugins/maven-invoker-plugin/run-mojo.html#pomExcludes
https://maven.apache.org/plugins/maven-invoker-plugin/examples/selector-scripts.html

**Comment by slawekjaranowski** [comment]
Cen we add GitHub actions in separate PR? - we will have status for next PR

**Comment by EagleErwin** [comment]
> Cen we add GitHub actions in separate PR? - we will have status for next PR

Do you mean _just_ adding the `.github/workflows/maven.yml` file in a separate Pull Request (and maybe the changed badge in README.md)? It will make the build fail for sure, but I'm not sure if that is actually what you want.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Verify
+    uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v3
+    with:
+      os-matrix: '[ "ubuntu-latest" ]'
+      jdk-matrix: '[ "8" ]'
```
Comment (slawekjaranowski) [inline .github/workflows/maven.yml:28]: please add 17

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      maven-matrix: '[ "3.6.3" ]'
```
Comment (slawekjaranowski) [inline .github/workflows/maven.yml:29]: please add 3.9.3 also

File: .github/workflows/maven.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      ff-jdk: '8'
```
Comment (slawekjaranowski) [inline .github/workflows/maven.yml:30]: use 17

File: .github/workflows/maven.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      ff-maven: '3.6.3'
```
Comment (slawekjaranowski) [inline .github/workflows/maven.yml:31]: use 3.9.3

File: pom.xml [hunk 6]
```diff
@@ -116,8 +116,8 @@
   </scm>
 
   <properties>
-    <mavenVersion>2.2.1</mavenVersion>
-    <mojo.java.target>1.6</mojo.java.target>
+    <mavenVersion>3.6.3</mavenVersion>
+    <mojo.java.target>1.8</mojo.java.target>
```
Comment (slawekjaranowski) [inline pom.xml:120]: should be `8`

**Comment by slawekjaranowski** [comment]
ok, we should merge whole PR at once ... please resolve my comments.

File: .github/workflows/maven.yml (hunk 2 above)
Comment (EagleErwin) [reply .github/workflows/maven.yml]: I noticed that '["8", "17"]' is [the default](https://github.com/apache/maven-gh-actions-shared/blob/v3/.github/workflows/maven-verify.yml#L38), so I removed the `jdk-matrix` line completely.

**Review by slachiewicz — APPROVED** [review]

File: .github/workflows/maven.yml (hunk 5 above)
Comment (EagleErwin) [reply .github/workflows/maven.yml]: Also the default, so I removed the line as well.

**Comment by slachiewicz** [comment]
Why have you changed from CentOS to Ubuntu?

**Comment by EagleErwin** [comment]
> Why have you changed from CentOS to Ubuntu?

I did it to match the build environment for Vagrant with the environment in GitHub CI. Would you prefer CentOS (9)?

**Comment by EagleErwin** [comment]
I processed all review comments. I needed to do some minor changes in some tests in order to be run flawless on all Java / Maven version combinations, but now the builds are [green](https://github.com/EagleErwin/rpm-maven-plugin/actions/runs/5621850390).

**Review by slachiewicz — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
@EagleErwin thanks - we have a green build.

**Pull request merged**